### PR TITLE
Introduce thin server definition

### PR DESCRIPTION
**Pull request by ignisf**
This commit introduces thin support out of the box.

It features a :Silent option to silence log output in the tests and makes sure
thin does not capture signals we want the testing framework to handle.

### CODE REVIEW

**Comment by ignisf** [comment]
Not sure how to handle the build failure on jruby... Should I just remove the dev dependency from the .gemspec and Gemfiles?

File: README.md [hunk 1]
```diff
@@ -97,13 +97,22 @@ Capybara.app = MyRackApp
 
 If you need to test JavaScript, or if your app interacts with (or is located at)
 a remote URL, you'll need to [use a different driver](#drivers).  If using Rails 5.0+, but not using the Rails system tests from 5.1, you'll probably also
-want to swap the "server" used to launch your app to Puma in order to match Rails defaults.
+want to swap the "server" used to launch your app to Puma or thin in order to match Rails defaults.
```
Comment (twalpole) [inline README.md]: `thin` is not the Rails default

**Comment by twalpole** [comment]
What is the motivation behind this?  My understanding is that `thin` has relatively low usage level when compared to puma/unicorn/etc. and from the unanswered issues and PRs it doesn't really seem to be maintained much anymore.  Why should we bake this into Capybara when it's easy enough for a user to register their own server if they want to use `thin`?

**Comment by ignisf** [comment]
Hello,

Just thought there should be a suggested way to use Capybara with Rails 5.2 and thin as it seems it's not that simple. I'm fine with this being documented somewhere in Capybara's docs, just so people do not rely on issue discussions (https://github.com/teamcapybara/capybara/issues/1713 for example).

File: README.md (hunk 1 above)
Comment (ignisf) [reply README.md]: Seems I did not understand the meaning of the sentence correctly.

**Comment by twalpole** [comment]
I'm not going to merge this as is, if you want to submit a PR that is purely documentation I think that would be more appropriate.

**Pull request closed without merging**